### PR TITLE
Still possibly overflow

### PR DESCRIPTION
**Pull request by mattn**
Add some fixes to https://github.com/vim/vim/pull/3999

### CODE REVIEW

**Comment by mattn** [comment]
I'll add some fix about local_bufsize.

File: src/dosinst.c [hunk 1]
```diff
@@ -506,7 +506,7 @@ uninstall_check(int skip_question)
     char	*uninstall_key = "software\\Microsoft\\Windows\\CurrentVersion\\Uninstall";
     char	subkey_name_buff[BUFSIZE];
     char	temp_string_buffer[BUFSIZE-2];
-    DWORD	local_bufsize = BUFSIZE;
+    DWORD	local_bufsize = sizeof(subkey_name_buff);
```
Comment (k-takata) [inline src/dosinst.c]: Same change needed to the line 529, 541 and 573.

File: src/dosinst.c [hunk 2]
```diff
@@ -1899,7 +1903,7 @@ build_shortcut(
     char	executable_path[BUFSIZE];
     char	link_name[BUFSIZE];
 
-    sprintf(executable_path, "%s\\%s", installdir, exename);
+    snprintf(executable_path, sizeof(executable_path), "%s\\%s", installdir, exename);
```
Comment (k-takata) [inline src/dosinst.c]: `snprintf` might not be able to use in old MSVC.
`_snprintf` might be used instead but the specification is slightly different.

File: src/dosinst.c [hunk 3]
```diff
@@ -1830,7 +1830,7 @@ create_shortcut(
 	    /* translate the (possibly) multibyte shortcut filename to windows
 	     * Unicode so it can be used as a file name.
 	     */
-	    MultiByteToWideChar(CP_ACP, 0, shortcut_name, -1, wsz, BUFSIZE);
+	    MultiByteToWideChar(CP_ACP, 0, shortcut_name, -1, wsz, sizeof(wsz));
```
Comment (k-takata) [inline src/dosinst.c]: Should be `sizeof(wsz) / sizeof(WCHAR)`.

File: src/dosinst.c [hunk 4]
```diff
@@ -521,12 +521,9 @@ uninstall_check(int skip_question)
 				     KEY_WOW64_64KEY | KEY_READ, &key_handle);
     CHECK_REG_ERROR(code);
 
-    for (key_index = 0;
-	 RegEnumKeyEx(key_handle, key_index, subkey_name_buff, &local_bufsize,
-		NULL, NULL, NULL, &temp_pfiletime) != ERROR_NO_MORE_ITEMS;
-	    key_index++)
+    key_index = 0;
+    while (TRUE)
```
Comment (k-takata) [inline src/dosinst.c:525]: It looks that you changed the logic.
With this change, `RegEnumKeyEx` won't executed before the first iteration of the loop.

**Pull request closed without merging**